File: editor/backend/releng/de.jabc.cinco.meta.core.parent/de.jabc.cinco.meta.plugin.pyro/frontend/app/lib/src/pages/logout/logout_component.dart
import 'package:angular/angular.dart';
import 'dart:html';
import 'dart:async';
import '../../service/base_service.dart';

@Component(
  selector: 'logout',
  directives: const [coreDirectives],
  templateUrl: 'logout_component.html',
)
class LogoutComponent implements OnInit {
  @override
  void ngOnInit() {
    logout(null);
  }

  void logout(dynamic e) {
    if (e != null) {
      e.preventDefault();
    }
    BaseService.logout().then((status) => {
          new Timer(new Duration(seconds: 2), () {
            window.location.href = BaseService.getBase() + "/#/home";
          })
        });
  }
}

File: services/main/frontend/src/app/features/home/pages/logout/logout.component.ts
import { Component, OnInit } from '@angular/core';
import { AppStoreService } from '../../../../core/services/stores/app-store.service';
import { ToastService, ToastType } from '../../../../core/services/toast.service';

@Component({
  selector: 'cc-logout',
  templateUrl: './logout.component.html'
})
export class LogoutComponent implements OnInit {

  constructor(private appStore: AppStoreService,
              private toastService: ToastService) {
  }

  ngOnInit(): void {
    this.appStore.logout().subscribe({
      next: () => this.toastService.show({
        message: 'You have been logged out.',
        type: ToastType.INFO
      }),
      error: console.error
    });
  }
}
